packages/flutter_tools/lib/src/build_runner/devfs_web.dart

```diff
@@ -834,7 +862,7 @@ class ReleaseAssetServer {
         'Content-Type': mimeType,
       });
     }
-    if (request.url.path == '') {
+    if (request.url.path == '' || enableIndexRewrite) {
```

Comment:
Is there any downside to always returning the index.html here?

Reply:
If this is the only change needed for all this plumbing, it seems like it would be easiest to always return the index, especially if this becomes the default in the future

Reply:
This is to make it optionally always return the index such that we don't break the previous behaviour.

Reply:
And it makes the release web server match the behaviour of the debug web server.